sandbox/lua/encoders/es_payload.lua

```diff
@@ -72,8 +72,16 @@ function process_message()
     if ts_from_message then
         ns = read_message("Timestamp")
     end
+
     local idx_json = elasticsearch.bulkapi_index_json(index, type_name, id, ns)
-    add_to_payload(idx_json, "\n", read_message("Payload"))
+    local payload  = read_message("Payload")
+
+    -- bulk api expects newline in the end of the payload
+    if not string.match(payload, "\n$") then
+	      payload = payload .. "\n"
+    end
```

Comment:
Since Lua strings are immutable, this does some unnecessary allocation and will likely cause unnecessary GC (see http://stackoverflow.com/questions/1405583/concatenation-of-strings-in-lua). You can avoid this by using `add_to_payload`:

```lua
local payload = read_message("Payload")
add_to_payload(idx_json, "\n", payload)
if not string.match(payload, "\n$") then
    add_to_payload("\n")
end
inject_payload()
```